nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-server/src/test/java/org/apache/nifi/distributed/cache/server/map/TestMapCacheClient.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.processor.Processor;
+import org.apache.nifi.remote.StandardVersionNegotiator;
+import org.apache.nifi.remote.VersionNegotiator;
+import org.apache.nifi.remote.io.socket.NetworkUtils;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.util.MockConfigurationContext;
+import org.apache.nifi.util.MockControllerServiceInitializationContext;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TestMapCacheClient {
+    private int port;
+    private EventServer server;
+    private final Serializer<String> stringSerializer = new StringSerializer();
+
+    @BeforeEach
+    public void setRunner() {
+        final TestRunner runner = TestRunners.newTestRunner(Mockito.mock(Processor.class));
+        port = NetworkUtils.getAvailableTcpPort();
+
+        final NettyEventServerFactory serverFactory = new NettyEventServerFactory(
+                InetAddress.getLoopbackAddress(), port, TransportProtocol.TCP);
+        serverFactory.setShutdownQuietPeriod(ShutdownQuietPeriod.QUICK.getDuration());
+        serverFactory.setShutdownTimeout(ShutdownQuietPeriod.QUICK.getDuration());
+        final ComponentLog log = runner.getLogger();
+        final VersionNegotiator versionNegotiator = new StandardVersionNegotiator(
+                ProtocolVersion.V3.value(), ProtocolVersion.V2.value(), ProtocolVersion.V1.value());
+
+        final MapCache mapCache = new SimpleMapCache("serviceIdentifier", 64, EvictionPolicy.FIFO) {
+            // override a service function that will cause client network timeout to fire
+            @Override
+            public MapPutResult put(ByteBuffer key, ByteBuffer value) throws IOException {
+                try {
+                    Thread.sleep(3000L);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                return super.put(key, value);
+            }
+        };
+
+        serverFactory.setHandlerSupplier(() -> Arrays.asList(
+                new CacheVersionResponseEncoder(),
+                new CacheOperationResultEncoder(),
+                new MapRemoveResponseEncoder(),
+                new MapSizeResponseEncoder(),
+                new MapValueResponseEncoder(),
+                new MapCacheRequestDecoder(log, 64, MapOperation.values()),
+                new MapCacheRequestHandler(log, mapCache),
```

Comment:
Are all of these Encoders required, or can this be scoped down to just the ones necessary for the test operation?

Reply:
Sure, I'll scope down.